### DISCUSSION TITLE
Pagination when listing image tags does not work

### DISCUSSION THREAD

**Discussion by alexey-igrychev** [community/community#25658] [Actions]
<p>I have container packages with thousands of tags, and I need the pagination to get them.</p>
<p>When I do the following request <code>ghcr.io/v2/&lt;name&gt;/tags/list?n=&lt;integer&gt;</code> (<a href="https://docs.docker.com/registry/spec/api/#listing-image-tags" rel="noopener nofollow ugc">Docker Registry HTTP API V2 method</a>) the response contains all tags (<code>n</code>) but no Link Header.</p>
<p>I think this is a serious problem because pagination is used in all registry tools (for instance, <a href="https://github.com/google/go-containerregistry/tree/main/cmd/crane" rel="noopener nofollow ugc">crane</a>).</p>

**Comment by alexey-igrychev** [comment]
<p>In addition, the same token is not enough when using the GitHub API to retrieve tags from package versions:</p>
<pre><code class="lang-auto">GET https://api.github.com/orgs/&lt;org_name&gt;/packages/container/&lt;package_name&gt;/versions?page=REDACTED&amp;per_page=REDACTED
{"message":"Resource not accessible by integration","documentation_url":"https://docs.github.com/rest/reference/packages#get-all-package-versions-for-a-package-owned-by-an-organization"}
</code></pre>

**Comment by gdomingu** [comment]
<p>Thank you very much for submitting this. Will follow up when the fix is out!</p>

**Comment by alexey-igrychev** [comment]
<p>Thanks. Without pagination, everything works fine, but for some users current behavior with pagination may be a surprise.</p>

**Comment by gdomingu** [comment, marked as answer]
<p>Alright this should be fixed now. Please let us know if there are still issues.</p>

**Comment by alexey-igrychev** [comment]
<p>Thanks for the quick fix, everything works properly.</p>
